programs/perpetuals/tests/anchor/test_client.ts

```diff
@@ -706,6 +708,88 @@ export class TestClient {
     }
   };
 
+  setCustomOraclePricePermissionless = async (
+    permissionlessOracleKeypair: Keypair,
+    price: number,
+    custody,
+    publishTime?,
+    noSignatureVerification?,
+    messageOverwrite?
+  ) => {
+    const schema = {
```

Comment:
are you sure anchor doesn't let you serialize directly?

Reply:
thanks for the push on this Alexey!  after some digging, yes we have access to the borsh serde that anchor uses.
```
this.program._coder.types.encode(
            "SetCustomOraclePricePermissionlessParams",
            setCustomOraclePricePermissionlessParams
          );
```
Updated to use the Anchor serializer instead, and removed the `borsh` .js dependency 👍